src/com/dabomstew/pkrandom/romhandlers/AbstractRomHandler.java

```diff
@@ -3182,25 +3182,67 @@ public void metronomeOnlyMode() {
 
     @Override
     public void customStarters(Settings settings) {
+        boolean abilitiesUnchanged = settings.getAbilitiesMod() == Settings.AbilitiesMod.UNCHANGED;
         int[] customStarters = settings.getCustomStarters();
         boolean allowAltFormes = settings.isAllowStarterAltFormes();
+        boolean banIrregularAltFormes = settings.isBanIrregularAltFormes();
 
         List<Pokemon> romPokemon = getPokemonInclFormes()
                 .stream()
                 .filter(pk -> pk == null || !pk.actuallyCosmetic)
                 .collect(Collectors.toList());
 
+        List<Pokemon> banned = getBannedFormesForPlayerPokemon();
         pickedStarters = new ArrayList<>();
-        Pokemon pkmn1 = romPokemon.get(customStarters[0]);
-        pickedStarters.add(pkmn1);
-        Pokemon pkmn2 = romPokemon.get(customStarters[1]);
-        pickedStarters.add(pkmn2);
+        if (abilitiesUnchanged) {
+            List<Pokemon> abilityDependentFormes = getAbilityDependentFormes();
+            banned.addAll(abilityDependentFormes);
+        }
+        if (banIrregularAltFormes) {
+            banned.addAll(getIrregularFormes());
+        }
+        // loop to add chosen pokemon to banned, preventing it from being a random option.
+        for (int i = 0; i < customStarters.length; i = i + 1){
+            if (!(customStarters[i] - 1 == 0)){
+                System.out.println(customStarters[i] - 1);
+                System.out.println(romPokemon.get(customStarters[i] - 1));
```

Comment:
Seems like you left some debug prints here. Other than that, everything looks good now

Reply:
resolved! thanks